File: Python/house-robber.py
class Solution:
    # @param nums, a list of integer
    # @return an integer
    def rob(self, nums):
        n = len(nums)
        dp = [0, 0]
        flag = 0
        for i in xrange(n):
            dp[flag] = max(dp[1-flag], dp[flag]+nums[i])
            flag = 1-flag
        return max(dp)

File: C++/house-robber.cpp
class Solution {
public:
    int rob(std::vector<int> &nums) {
        int dp[2] = {0, 0};
        int flag = 0;
        for(int i=0; i<nums.size(); ++i) {
            dp[flag] = std::max(dp[1-flag], dp[flag]+nums[i]);
            flag = 1-flag;
        }
        return std::max(dp[0], dp[1]);
    }
};
